src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -586,6 +586,7 @@ The .NET Foundation licenses this file to you under the MIT license.
       <CustomLinkerArg Include="-Wl,--global-base=$(IlcWasmGlobalBase)" />
       <CustomLinkerArg Include="-Wl,-z,stack-size=$(IlcWasmStackSize)" />
       <CustomLinkerArg Include="-mexec-model=reactor" Condition="'$(NativeLib)' == 'Shared'" />
+      <CustomLinkerArg Include="-Wl,--component-type,&quot;$(IlcSdkPath.Replace(&quot;\&quot;, &quot;/&quot;))WasiHttpWorld_component_type.wit&quot;" />
```

Comment:
I am assuming that the produces component here will only have imports that it actually needs/uses (so it is, in effect, harmless to add WITs for as many things as we would deem our "baseline OS" for WASI). Is that correct?

Reply:
Yes, that's correct.  `wasm-component-ld` will only use the parts it needs.  The .o files to be linked are what determine what actually gets imported and/or exported.